### PR TITLE
Allow to use the package with Python 3.12.

### PR DESCRIPTION
**Pull request by icemac**
See #246.

### CODE REVIEW

**Comment by icemac** [comment]
I have no idea why Windows 3.12 fails: https://github.com/zopefoundation/RestrictedPython/actions/runs/4351559170/jobs/7603343628

**Review by dataflake — APPROVED** [review]

**Comment by icemac** [comment]
I created https://github.com/pypa/pip/issues/11844 for the Windows 3.12 issue.

File: tests/test_compile.py [hunk 1]
```diff
@@ -102,7 +103,13 @@ def test_compile__compile_restricted_exec__5():
...
     assert result.warnings == []
     assert result.used_names == {}
-    assert result.errors == ('source code string cannot contain null bytes',)
+    if IS_PY312_OR_GREATER:
+        assert result.errors == (
+            'Line None: SyntaxError: source code string cannot contain null'
+            ' bytes at statement: None',)
+    else:
+        assert result.errors == (
+            'source code string cannot contain null bytes',)
```
Comment (mauritsvanrees) [inline tests/test_compile.py:112]: This should fit all Python versions, if it is okay that we do not test the exact error message:

```suggestion
    assert 'source code string cannot contain null bytes' in result.errors
```

Comment (icemac) [reply tests/test_compile.py:112]: `result.errors` is a tuple, so it is not that easy. I'll keep my version.

**Comment by icemac** [comment]
I just removed the `pip list` call which is not absolutely necessary. Now the tests are green.

Thank you for reviewing this PR. 😃

**Pull request merged**